### PR TITLE
azurerm_devspace_controller: deprecate sku in favour of sku_name

### PR DESCRIPTION
**Pull request by katbyte**
partially addresses #1500

### CODE REVIEW

File: website/docs/guides/2.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -253,6 +253,10 @@ Azure Container Service (ACS) is being Deprecated in favour of Azure Kubernetes
 
 The deprecated `failover_policy` block will be removed. This has been replaced by the `geo_location` block.
 
+### Resource: `azurerm_devspace_controller`
+
+The deprecated `sku` block has been simplified and replaced by the `sku_name` property and will be removed.
```
Comment (tracypholmes) [inline website/docs/guides/2.0-upgrade-guide.html.markdown]:
```suggestion
The deprecated `sku` block has been replaced by the `sku_name` field and will be removed.

The `sku_name` field will become Required.
```

File: website/docs/guides/2.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -253,6 +253,10 @@ Azure Container Service (ACS) is being Deprecated in favour of Azure Kubernetes
...
+
 ### Resource: `azurerm_ddos_protection_plan`
 
 The `azurerm_ddos_protection_planr` resource will be deprecated in favour of a new resources `azurerm_network_ddos_protection_plan`.
```
Comment (tracypholmes) [inline website/docs/guides/2.0-upgrade-guide.html.markdown]:
```suggestion
The `azurerm_ddos_protection_plan` resource will be deprecated in favour of a new resources `azurerm_network_ddos_protection_plan`.
```

File: azurerm/internal/services/devspace/resource_arm_devspace_controller.go [hunk 3]
```diff
@@ -47,25 +47,37 @@ func resourceArmDevSpaceController() *schema.Resource {
...
+				Deprecated:    "This property has been deprecated in favour of the 'sku_name' property and will be removed in version 2.0 of the provider",
+				MaxItems:      1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
 							Type:     schema.TypeString,
 							Required: true,
-							ForceNew: true,
 							ValidateFunc: validation.StringInSlice([]string{
 								"S1",
```
Comment (WodansSon) [inline azurerm/internal/services/devspace/resource_arm_devspace_controller.go]: For consistency with the other `sku_name`s should we make this `S_1`?

Comment (katbyte) [reply azurerm/internal/services/devspace/resource_arm_devspace_controller.go]: I don't want to make confusing alternatives, its up to azure to change it

**Pull request merged**